#  <p style="text-align: center;">Практическое задание к Семинару 1.<br><br>Урок 1. Расчет вероятности случайных событий.</p>


In [2]:
from math import factorial

m = 715


***Задание 1.*** Из колоды в 52 карты извлекаются случайным образом 4 карты.<br>
```
a) Найти вероятность того, что все карты – крести.
б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз.
```

<font size=4.5 color=#3ac444>**РЕШЕНИЕ:**</font>

a) *Найти вероятность того, что все карты – крести.*
1) Находим все сочетания из 4 карт возможные в колоде из 52 карт.
2) Находим все сочетания 4 карт одной масти (учитываем, что в колоде 4 масти, тогда $\frac{52}{4}=13$).
3) Находим вероятность вытащить 4 карты одной масти.

Формула вероятности:  
$\displaystyle P(A)=\frac{m}{n}$, где $m$-число благоприятных исходов, $n$-общее число всех исходов.


Формула подсчета количества сочетаний по $k$ элементов из множества $n$:  
$\displaystyle C^k_n=\frac{n!}{k!(n-k)!}$ 

, тогда  все сочетания из 4 карт возможные в колоде из 52 карт: $\displaystyle n=C^4_{52}$,<br>, а все сочетания 4 карт одной масти (учитываем, что в колоде 4 масти, тогда $\frac{52}{4}=13$ ): $\displaystyle m=C^4_{13}$

In [28]:
def combinations(n, k): #функция расчета количества сочетаний
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

In [29]:
# n - все возможные сочетания
n=combinations(52, 4)
print(f'n = {n}')

n = 270725


In [30]:
# m - все сочетания одной масти
m=combinations(13, 4)
print(f'm = {m}')

m = 715


In [31]:
P=(m/n)
P

0.0026410564225690276

б) *Найти вероятность, что среди 4-х карт окажется хотя бы один туз.*
1) Находим все сочетания из 4 карт возможные в колоде из 52 карт.  => $\displaystyle n=C^4_{52}$
2) Находим все сочетания 4 карт, в которых нет ни одного туза. (в колоде 4 туза разной масти, тогда 52-4=48) => $\displaystyle n=C^4_{48}$
3) Находим вероятность вытащить хотябы один туз. Учтем,что сумма вероятностей противоположных событий равна 1:<br>  $\displaystyle P(A)+P(\bar{A})=1$, т.е. количество способов извлечь 4 карты из колоды **без** тузов. Тогда $\displaystyle P(A)=1-P(\bar{A})$

In [32]:
# n - все возможные сочетания
n=combinations(52, 4)
print(f'n = {n}')

n = 270725


In [33]:
# m - все сочетания одной масти
m=combinations(48, 4)
print(f'm = {m}')

m = 194580


In [34]:
P_tyz=1-(m/n)
P_tyz

0.2812632745405854

<div class="alert alert-block alert-success"><font size=3 color='b'><b>ОТВЕТ: </b><br>  1.а) Вероятность вытащить все карты масти крести равна $P\approx 0.0026 \approx 0.26\%$.<br>1.б) Вероятность, что среди 4-х карт окажется хотя бы один туз равна $P\approx 0.2813 \approx 28.13\%$.</font></div>

<br><br><br><font size=10 color=#32CD32><b>-------------------------------------------------------------</b></font>

#  <p style="text-align: center;">ПРОШУ НЕ ОЦЕНИВАТЬ ДО 14.10.2023.<br><br>ОБЯЗУЮСЬ ДОДЕЛАТЬ ВСЕ ЗАДАНИЯ до 22:00 (МСК) 14.10.2023<br><br>*нахожусь в командировке, не всегда есть возможность сделать задания*</p>


<br><br><br><font size=10 color=#32CD32><b>-------------------------------------------------------------</b></font>